SDK v2 migration guide.md

```diff
@@ -0,0 +1,399 @@
+| DeviceClient#startDeviceTwin(IotHubEventCallback, Object, TwinPropertyCallback, Object); | DeviceClient#subscribeToDesiredPropertiesAsync(DesiredPropertiesSubscriptionCallback, Object, DesiredPropertiesCallback, Object);  |
+| DeviceClient#startDeviceTwin(IotHubEventCallback, Object, TwinPropertiesCallback, Object); | DeviceClient#subscribeToDesiredPropertiesAsync(DesiredPropertiesSubscriptionCallback, Object, DesiredPropertiesCallback, Object);  |
+| DeviceClient#startDeviceTwin(IotHubEventCallback, Object, PropertyCallBack<Type1, Type2>, Object); | DeviceClient#subscribeToDesiredPropertiesAsync(DesiredPropertiesSubscriptionCallback, Object, DesiredPropertiesCallback, Object);  |
+| DeviceClient#getDeviceTwin(); | DeviceClient#getTwinAsync();  |
+| DeviceClient#subscribeToDesiredProperties(Map<Property, Pair<PropertyCallBack<String, Object>, Object>>); | DeviceClient#subscribeToDesiredPropertiesAsync(DesiredPropertiesSubscriptionCallback, Object, DesiredPropertiesCallback, Object);  |
+| DeviceClient#subscribeToTwinDesiredProperties(Map<Property, Pair<TwinPropertyCallBack, Object>>(); | DeviceClient#subscribeToDesiredPropertiesAsync(DesiredPropertiesSubscriptionCallback, Object, DesiredPropertiesCallback, Object);  |
+| DeviceClient#sendReportedProperties(Set<Property>); | DeviceClient#updateReportedPropertiesAsync(TwinCollection, ReportedPropertiesCallback, Object);  |
+| DeviceClient#sendReportedProperties(Set<Property>, int); | DeviceClient#updateReportedPropertiesAsync(TwinCollection, ReportedPropertiesCallback, Object);  |
+| DeviceClient#sendReportedProperties(ReportedPropertiesParameters); | DeviceClient#updateReportedPropertiesAsync(TwinCollection, ReportedPropertiesUpdateCorrelatingMessageCallback, Object);  |
+| DeviceClient#sendReportedProperties(Set<Property>, Integer, CorrelatingMessageCallback, Object, IotHubEventCallback, Object); | DeviceClient#updateReportedPropertiesAsync(TwinCollection, ReportedPropertiesUpdateCorrelatingMessageCallback, Object);  |
+| DeviceClient#setOption(String, Object); | no equivalent method***  |
+| DeviceClient#setProxySettings(ProxySettings); | no equivalent method****  |
+
+** This method has been split into the three individual steps that this method used to take. See the file upload samples within this repo for an example of how to do file upload using these discrete steps.
+
+*** The options that were previously set in this method are now set at DeviceClient constructor time in the optional ClientOptions parameter.
+
+**** Proxy settings are now set at ModuleClient constructor time in the optional ClientOptions parameter,
+
+#### ModuleClient
+
+| V1 class#method  | Equivalent V2 class#method |
+|:---|:---|
+| ModuleClient#createFromEnvironment(); | ModuleClient#createFromEnvironment(UnixDomainSocketChannel);  |
```

Comment:
Q - do you think it is worth calling out why the default constructor is now taking in the unix socket channel information? (here, and in other APIs as well)?

I think if someone looks at this document as the starting point, then it might be confusing to simply compare the APIs with not much additional information. However, if someone is coming to this document from code (which I think we expect most people to be doing), then it might not be as confusing since our doc comments would have called out the reasoning for the params being required.

Thought?

Reply:
Good point. I'll add an asterisk on these cases and explain more beneath the table about this particular change